### PR TITLE
Docs page on interoperability

### PR DESCRIPTION
**Pull request by TomNicholas**
Builds upon #7991 by adding a page to the internals enumerating all the different ways in which xarray is interoperable.

Would be nice if https://github.com/pydata/xarray/pull/6975 were merged so that I could link to it from this new page.

- [x] Addresses comment in https://github.com/pydata/xarray/pull/6975#issuecomment-1246487152
- [ ] ~~Tests added~~
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] ~~New functions/methods are listed in `api.rst`~~

### CODE REVIEW

File: doc/internals/internal-design.rst
Comment (Illviljan) [inline doc/internals/internal-design.rst:64]:
```suggestion
- ``attrs``: A dictionary of metadata associated with this array. By
  convention, xarray's built-in operations never use this metadata.
- ``encoding``: Another dictionary used to store information about how
```
If I recall correctly attrs and encoding are simple dicts, we don't do anything fancy with them.

Not important but an example where we use `attrs` directly is to add units to plots with `attrs["units"]`.

Comment (TomNicholas) [reply doc/internals/internal-design.rst:64]: I think you're right. (I actually didn't write this section, just moved it, so I wonder if it changed at some point...)

File: doc/internals/interoperability.rst
Comment (max-sixty) [inline doc/internals/interoperability.rst:20]:
```suggestion
    One obvious way in which xarray could be more flexible is that whilst subclassing xarray objects is possible, we
    generally advise against it, instead recommending composition over inheritance. See the
```
```suggestion
    One obvious way in which xarray could be more flexible is that whilst subclassing xarray objects is possible, we
    currently don't support it in most transformations, instead recommending composition over inheritance. See the
```

- I have had success using it as a kinda "central object", with methods that operate on the dataset — which works quite well as long as no transformations are done 
- I would change "generally" to "currently" to highlight that it could be possible in the future...

File: doc/internals/interoperability.rst
Comment (max-sixty) [inline doc/internals/interoperability.rst:1]: Really cool page!

**Review by max-sixty — APPROVED** [review]
Very nice!

**Comment by max-sixty** [comment]
This just came up on my notifications — looks great — shall we merge?

**Comment by TomNicholas** [comment]
You must be getting a lot of notifications if you got one just because I committed here! 
 :exploding_head: 

Yes great I was just waiting for an approval - will merge shortly.

File: doc/internals/interoperability.rst
Comment (TomNicholas) [reply doc/internals/interoperability.rst:20]: I'll just add a note with a link to issue https://github.com/pydata/xarray/issues/3980

File: doc/internals/interoperability.rst
Comment (TomNicholas) [inline doc/internals/interoperability.rst:7]:
```suggestion
Making xarray as flexible as possible is the common theme of most of the goals on our :ref:`roadmap`.
```

File: doc/internals/interoperability.rst
Comment (TomNicholas) [reply doc/internals/interoperability.rst]: Oh I already link to it! I'll change the wording to be more encouraging to developers interested in helping.

File: doc/internals/interoperability.rst
Comment (TomNicholas) [inline doc/internals/interoperability.rst:22]:
```suggestion
    :ref:`internal design page <internal design.subclassing>` for the rationale and look at the corresponding `GH issue <https://github.com/pydata/xarray/issues/3980>`_
    if you're interested in improving support for subclassing!
```

**Pull request merged**
